File: C/cryptography/MonoAlphabeticCipher.c
/* Monoalphabetic substitution cipher basically replaces each letter of a plaintext by another letter or symbol, depending only on the letter.
   The same plaintext letter is always replaced by the same ciphertext letter and the same letter can be replaced by letter itself.
   Example if a is there we replace it with B to Z and so on for the others.*/

#include<stdio.h>
#include<string.h>

//this is the encription function
void monoalphebaticcipher(char input[100],int size){
     
     char output[size-1]; 
     char cipherer[26] = {'Z','Y','X','W','V','U','T','S','R','Q','P','O','N','M','L','K','J','I','H','G','F','E','D','C','B','A'};     
     char list[26] = {'a', 'b', 'c', 'd', 'e', 'f', 'g', 'h', 'i', 'j', 'k', 'l', 'm', 'n', 'o','p', 'q', 'r', 's', 't', 'u', 'v', 'w', 'x', 'y', 'z'};  
     char space = ' ';
     //iterate through input and cipher the plane text as output
     for(int i=0;i<size-1;i++){     
         for(int j=0;j<26;j++){
             if(input[i]==list[j])
                output[i] = cipherer[j];
             else if( input[i] == space)
                output[i] = space;    
         } 
     } 
  
     printf("The Cipher text :");
     for(int j=0;j<size-1;j++)
         printf("%c",output[j]);  
}

int main(void){

     char input[100];
     printf("Give the plain text here in small letters :");
     //Taking input from user
     fgets(input,100,stdin);
     int size = strlen(input);
     //calling monoalphabeticcipher
     monoalphebaticcipher(input,size);
     return 0;
}
/*
  Sample I/O :
  Give the plain text here in small letters :hello world
  The Cipher text :SVOOL DLIOW
*/

File: Python/cryptography/monoalphabeticcipher.py
''' Monoalphabetic substitution cipher basically replaces each letter of a plaintext by another
    letter or symbol, depending only on the letter. The same plaintext letter is always replaced
    by the same ciphertext letter and the same letter can be replaced by letter itself.
    Example if a is there we replace it with B to Z .....so on.'''

'''This is the encryption function which takes planetext
   as input and returns us the ciphertext list.''' 
def monoalphabeticcipher(plaintext,size):
   
   output = ['']*size
   #This is the cipher list we use to encrypt the message 
   cipherer = ['Z','Y','X','W','V','U','T','S','R','Q','P','O','N','M','L','K','J','I','H','G',
   'F','E','D','C','B','A'];
   list = ['a', 'b', 'c', 'd', 'e', 'f', 'g', 'h', 'i', 'j', 'k', 'l', 'm', 'n', 'o','p', 'q',
   'r', 's', 't', 'u', 'v', 'w', 'x', 'y', 'z']
   
   space = ' '   
   
   '''We iterate through the plaintext given to us an find out
      what letter it is and then we use the list called cipherer
      to replace that letter. We keep the space as space in output'''              
   for i in range(size):   
      for j in range(26):
        if plaintext[i] == list[j] :
           output[i] = cipherer[j];
        elif plaintext[i] == space :
           output[i] = space;    
   
   #return the encrypted message
   return output
  
if __name__ == '__main__':
   
   #we take the plaintext input here
   plaintext = input("Enter the plaintext here :")
   size = len(plaintext)
  
   '''calling the monoalphabeticcipher function on the given plaintext
      to return the encrypted message'''
   encrypted_message = monoalphabeticcipher(plaintext,size)
   
   #printing the ciphertext we got using the monoalphabetic function
   print("Encripted plaintext is :","".join(encrypted_message))

'''
   Sample I/O :

   a)Case where plaintext has space :
    
   Enter the plaintext here :hello world
   Encripted plaintext is : SVOOL DLIOW 

   b)Case where plaintext dosen't have space :

   Enter the plaintext here :python
   Encripted plaintext is : KBGSLM    
'''

